datalake/src/main/java/com/sequenceiq/datalake/service/sdx/dr/SdxBackupRestoreService.java

```diff
@@ -331,6 +332,30 @@ private AttemptResult<StackV4Response> getStackResponseAttemptResult(SdxCluster
         }
     }
 
+    /**
+     * Get which step the program is at when the backup/restore operation fails.
+     *
+     * @param sdxId The ID of the cluster.
+     * @return A string of the information about where the operation fails.
+     */
+    public String createDatabaseBackupRestoreErrorStage(Long sdxId) {
+        SdxCluster sdxCluster = sdxClusterRepository.findById(sdxId).orElseThrow(notFound("SDX cluster", sdxId));
```

Comment:
you should use `SdxSercvice#getById` instead of this

Reply:
@lacikaaa Updated to use SdxSercvice#getById. Thank you a lot for letting me know about this!